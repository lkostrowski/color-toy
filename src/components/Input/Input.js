import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import colors from "../../colors";

const InputContainer = glamorous.input({
    display: 'block',
    width: '100%',
    height: '3rem',
    background: colors.light,
    outline: 'none',
    border: `1px solid ${colors.lightGray}`,
    fontSize: '1.8rem',
    color: colors.midGray,
    letterSpacing: '0.05em',
});

const Input = ({...rest}) => (
    <InputContainer {...rest} />
);

Input.propTypes = {}

export default Input;
