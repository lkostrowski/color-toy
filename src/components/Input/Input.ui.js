import glamorous from "glamorous";
import colors from "../../colors";

export const InputContainer = glamorous.input({
    display: 'block',
    width: '100%',
    height: '3rem',
    background: colors.white,
    outline: 'none',
    border: `1px solid ${colors.lightGray}`,
    fontSize: '1.8rem',
    color: colors.midGray,
    letterSpacing: '0.05em',
    padding: '0 1rem'
});
