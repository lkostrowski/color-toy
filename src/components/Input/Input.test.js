import Input from './Input';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Input component', () => {
    it('render expected html', () => {
        const markup = renderer.create(
            <Input type="text"/>
        ).toJSON();

        expect(markup).toMatchSnapshot();
    });
});
