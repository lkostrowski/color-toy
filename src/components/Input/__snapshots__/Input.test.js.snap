// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component render expected html 1`] = `
<input
  className="css-1qyn3bs"
  type="text"
/>
`;
